### ISSUE TITLE
citations whose id starts with a digit are not processed

### ISSUE THREAD

**Issue by mcmtroffaes** [jgm/pandoc-citeproc#97]
Example:

``` {.markdown}
---
title: Test
references:
- id: 1657:huygens
  publisher-place: Lugd. Batav.
  page: 517-524
  container-title: 'Exercitationum mathematicarum libri quinque: Quibus accedit
    Christiani Hugenii tractatus de ratiociniis in aleæ ludo'
  author:
  - family: Huygens
    given: Christiaan
  issued:
    year: 1657
  title: De ratiociniis in ludo aleæ
  type: chapter
  publisher: Ex officina Johannis Elsevirii
  editor:
  - family: Schooten
    given: Franciscus á
...

[@1657:huygens]
```

renders - without error messages - as

```
[@1657:huygens]
```

instead of

```
(Huygens 1657)

Huygens, Christiaan. 1657. “De Ratiociniis in Ludo Aleæ.” In Exercitationum Mathematicarum Libri Quinque: Quibus Accedit Christiani Hugenii Tractatus de Ratiociniis in Aleæ Ludo, edited by Franciscus á Schooten, 517–24. Lugd. Batav.: Ex officina Johannis Elsevirii.
```

Changing the id to, for example, `x1657:huygens`, makes it work again, so the issue appears to be that the id starts with a digit.

In bibtex, it is perfectly valid for citation ids to start with the year, so it would be nice if pandoc-citeproc could support this as well.

**Comment by jgm** [comment]
That's intentional (see pandoc User's Guide under citations).

I worried that if we allowed citations to start with digits, we'd
falsely capture some things that weren't intened as citations,
e.g. `15 apples @5ea`.  Maybe this is overthinking, though.


+++ Matthias C. M. Troffaes [Jan 09 15 01:50 ]:

**Comment by mcmtroffaes** [comment]
Thanks for your quick response. Indeed this is mentioned in the User's Guide. I thought I had read it, but I missed this detail.

Nevertheless, personally, I find the distinction between digits and letters surprising. The standard way in markdown to prevent special characters to act their magic is to backslash them. This already works e.g. `\@x1657:huyghens` is indeed rendered correctly as `@x1657:huyghens` and not as a citation. The same could work for `\@5ae`.

A change would of course break backwards compatibility. It's hard to tell how many people would rely on `@5ae` not being a citation in documents that use citeproc; probably not many?

**Comment by jgm** [comment]
Closed by jgm/pandoc#1954.
